bin/p2-label/main.go

```diff
@@ -0,0 +1,169 @@
+		}
+		fmt.Printf("%s/%s: %s\n", labelType, *showID, labelsForEntity.Labels.String())
+		return
+	case CmdApply:
+		autoConfirm = *applyAutoConfirm
+
+		labelType, err := labels.AsType(*applyLabelType)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unrecognized type %s. Check the commandline and documentation.\nhttps://godoc.org/github.com/square/p2/pkg/labels#pkg-constants\n", *applyLabelType)
+			break
+		}
+
+		subject, err := klabels.Parse(*applySubjectSelector)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing subject label. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		additive := klabels.Set(*applyAddititiveLabels)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing additive label set. Check the syntax.\n%v\n", err)
+			break
+		}
+		destructive := klabels.Set(*applyDestructiveLabels)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing destructive label set. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		cachedMatch := false
+		matches, err := applicator.GetMatches(subject, labelType, cachedMatch)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while finding label matches. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		additiveSelector := additive.AsSelector()
+		destructiveSelector := destructive.AsSelector()
+
+		if !additiveSelector.Empty() {
+			fmt.Printf("labels to be added: %s\n", additive.String())
+		}
+
+		if !destructiveSelector.Empty() {
+			fmt.Printf("labels to be removed: %s\n", destructive.String())
+		}
+
+		var labelsForEntity labels.Labeled
+		for _, match := range matches {
+			entityID := match.ID
+
+			err := applyLabels(applicator, entityID, labelType, additiveSelector, destructiveSelector)
+			if err != nil {
+				fmt.Printf("Encountered err during labeling, %v", err)
+				exitCode = 1
+			}
+
+			labelsForEntity, err = applicator.GetLabels(labelType, entityID)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Got error while querying labels. %v\n", err)
+				exitCode = 1
+				continue
+			}
+			fmt.Printf("%s/%s: %s\n", labelType, entityID, labelsForEntity.Labels.String())
+		}
+		break
+	}
+
+	os.Exit(exitCode)
+}
+
+func applyLabels(applicator labels.Applicator, entityID string, labelType labels.Type, additiveLabels, destructiveLabels klabels.Selector) error {
+	var err error
+	if !confirm(fmt.Sprintf("mutate the labels for this entity %s/%s", labelType, entityID)) {
+		return nil
+	}
+	if !additiveLabels.Empty() {
+		for _, label := range strings.Split(additiveLabels.String(), ",") {
+			kvs := strings.Split(label, "=")
+			if len(kvs) != 2 {
+				return util.Errorf("Unexpected non-binary label")
+			}
+			err = applicator.SetLabel(labelType, entityID, kvs[0], kvs[1])
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error while appyling label. k/v: %s/%s. \n%v\n", kvs[0], kvs[1], err)
+			}
+		}
+	}
+	if !destructiveLabels.Empty() {
+		for _, label := range strings.Split(destructiveLabels.String(), ",") {
+			kvs := strings.Split(label, "=")
```

Comment:
hmmmmmm, this requires some thought. The value is going to be ignored so it's immaterial but the label parser likes things to look like `k=v`.